File: app/src/main/java/com/senerunosoft/puantablosu/ui/home/LatestGameFragment.java
package com.senerunosoft.puantablosu.ui.home;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.fragment.NavHostFragment;
import com.senerunosoft.puantablosu.IGameService;
import com.senerunosoft.puantablosu.R;
import com.senerunosoft.puantablosu.databinding.FragmentLatestGameBinding;
import com.senerunosoft.puantablosu.model.Game;
import com.senerunosoft.puantablosu.service.GameService;
import com.senerunosoft.puantablosu.viewmodel.GameViewModel;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class LatestGameFragment extends Fragment {

    FragmentLatestGameBinding binding;
    SharedPreferences sharedPreferences;
    IGameService gameService;
    GameViewModel gameViewModel;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentLatestGameBinding.inflate(inflater, container, false);
        View root = binding.getRoot();
        return root;
    }

    @Override
    public void onViewCreated(@NonNull @NotNull View view, @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        defVariable();
        getSharedElements();
    }

    private void defVariable() {
        gameService = new GameService();
        sharedPreferences = requireActivity().getSharedPreferences("game", Context.MODE_PRIVATE);
        gameViewModel = new ViewModelProvider(requireActivity()).get(GameViewModel.class);
    }

    private void getSharedElements() {
        String gameIds = sharedPreferences.getString("gameIds", "");
        if (gameIds.isEmpty()) {
//            binding.textView.setText("Oynanmış oyun bulunmamaktadır.");
        } else {
            getGameNameList(gameIds);
        }

    }

    private void getGameNameList(String gameIds) {
        List<Game> gameList = new ArrayList<>();
        List<String> gameIdList = Arrays.asList(gameIds.split(","));
        for (String gameId : gameIdList) {
            String gameJson = sharedPreferences.getString(gameId, "");
            if (!gameJson.isEmpty()) {
                Game game = gameService.deserializeGame(gameJson);
                gameList.add(game);
            }
        }

        List<String> gameNameList = new ArrayList<>();
        for (Game game : gameList) {
            gameNameList.add(game.getGameTitle());
        }
        ArrayAdapter<String> adapter = new ArrayAdapter<>(requireContext(), android.R.layout.simple_list_item_1, gameNameList);
        binding.latestGameList.setAdapter(adapter);

        binding.latestGameList.setOnItemClickListener((parent, view, position, id) -> {
            Game game = gameList.get(position);
            gameViewModel.setGameInfo(game);
            NavHostFragment.findNavController(LatestGameFragment.this)
                    .navigate(R.id.boardScreenFragment);
        });


    }


    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/senerunosoft/puantablosu/ui/home/LatestGameFragment.kt
package com.senerunosoft.puantablosu.ui.home

import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.fragment.NavHostFragment
import com.senerunosoft.puantablosu.IGameService
import com.senerunosoft.puantablosu.R
import com.senerunosoft.puantablosu.databinding.FragmentLatestGameBinding
import com.senerunosoft.puantablosu.model.Game
import com.senerunosoft.puantablosu.service.GameService
import com.senerunosoft.puantablosu.viewmodel.GameViewModel

class LatestGameFragment : Fragment() {

    private var _binding: FragmentLatestGameBinding? = null
    private val binding get() = _binding!!
    
    private lateinit var sharedPreferences: SharedPreferences
    private lateinit var gameService: IGameService
    private lateinit var gameViewModel: GameViewModel

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = FragmentLatestGameBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        defineVariables()
        getSharedElements()
    }

    private fun defineVariables() {
        gameService = GameService()
        sharedPreferences = requireActivity().getSharedPreferences("game", Context.MODE_PRIVATE)
        gameViewModel = ViewModelProvider(requireActivity())[GameViewModel::class.java]
    }

    private fun getSharedElements() {
        val gameIds = sharedPreferences.getString("gameIds", "") ?: ""
        if (gameIds.isEmpty()) {
            // binding.textView.text = "Oynanmış oyun bulunmamaktadır."
        } else {
            getGameNameList(gameIds)
        }
    }

    private fun getGameNameList(gameIds: String) {
        val gameList = mutableListOf<Game>()
        val gameIdList = gameIds.split(",")
        
        for (gameId in gameIdList) {
            val gameJson = sharedPreferences.getString(gameId, "") ?: ""
            if (gameJson.isNotEmpty()) {
                gameService.deserializeGame(gameJson)?.let { game ->
                    gameList.add(game)
                }
            }
        }

        val gameNameList = gameList.map { it.gameTitle }
        val adapter = ArrayAdapter(requireContext(), android.R.layout.simple_list_item_1, gameNameList)
        binding.latestGameList.adapter = adapter

        binding.latestGameList.setOnItemClickListener { _, _, position, _ ->
            val game = gameList[position]
            gameViewModel.setGameInfo(game)
            NavHostFragment.findNavController(this)
                .navigate(R.id.boardScreenFragment)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}